generator/generator.go

```diff
@@ -290,40 +301,43 @@ func (g Generator) Generate(w io.Writer) error {
 var errInterfaceNotFound = errors.New("interface type declaration not found")
 
 // findInterface looks for the interface declaration in the given directory
-// and returns a list of the interface's methods and a list of imports from the file
+// and returns the generic params if exists, a list of the interface's methods, and a list of imports from the file
 // where interface type declaration was found
-func findInterface(fs *token.FileSet, currentPackage *packages.Package, p *ast.Package, interfaceName string) (methods methodsList, imports []*ast.ImportSpec, err error) {
-	var found bool
-	var types []*ast.TypeSpec
-	var it *ast.InterfaceType
-
+func findInterface(fs *token.FileSet, currentPackage *packages.Package, p *ast.Package, interfaceName string, genericParams genericsParams) (genericsTypes genericsTypes, methods methodsList, imports []*ast.ImportSpec, err error) {
```

Comment:
Maybe it's time to create request/response structs for this method?

Reply:
@hexdigest the name also is not good, because this function has N responsabilities.